python/tests/catalog/test_glue.py

```diff
@@ -52,7 +52,7 @@ def test_create_table_with_database_location(
     _bucket_initialize: None, _patch_aiobotocore: None, table_schema_nested: Schema, database_name: str, table_name: str
 ) -> None:
     identifier = (database_name, table_name)
-    test_catalog = GlueCatalog("glue")
+    test_catalog = GlueCatalog("glue", **{"py-io-impl": "pyiceberg.io.fsspec.FsspecFileIO"})
```

Comment:
sorry just to get some more context here, why do we need to use `FsspecFileIO` for the tests? Would `PyArrowFileIO` not work?

Reply:
I think `PyArrowFileIO` is not compatible with the mocked S3 FIleSystem (handled by `moto` and `_patch_aiobotocore`) we used in the unit test. I received 
```
OSError: When getting information for key '...' in bucket 'test_bucket': AWS Error ACCESS_DENIED during HeadObject operation: No response body.
```
when using `PyArrowFileIO` in the unit test. Seems it still trying to interact with the real s3 bucket not the mocked one.

I also rerun the `integration_test_glue.py` with the new changes and verify that everything works fine with io default to `PyArrowFileIO`

Reply:
Thanks for the explanation!

Reply:
Thanks for jumping in here @JonasJ-ap 👍🏻